### PR TITLE
Merging guidelines

### PR DESCRIPTION
**Pull request by macpijan**

### CODE REVIEW

File: docs/dev-proc/source-code-structure.md [hunk 1]
```diff
@@ -38,4 +38,48 @@ with following exceptions:
...
+1. Review the code in GitHub.
+1. Make sure to receive at least one `Approve` in the review process.
+1. Make sure that **all** change requests are resolved.
+1. Merge the branch using git CLI. In case of merging the `feature` branch into
+   `develop` branch it may look as follows:
+
+    ```bash
+    $ git fetch dasharo
+    $ git checkout dasharo/<platform>/develop
+    $ git checkout -b <platform>/develop
```
Comment (krystian-hebel) [inline docs/dev-proc/source-code-structure.md:64]: These two commands can be done as one.

**Review by krystian-hebel — APPROVED** [review]
Tested CLI commands, work as expected.

File: docs/dev-proc/source-code-structure.md [hunk 2]
```diff
@@ -38,4 +38,48 @@ with following exceptions:
...
-Force-pushes to other branches are unconditionally forbidden.
+Force-pushes to  `<platform>/release` branches are unconditionally forbidden.
+
+## Merging guidelines
+
+We want to keep the history linear. The `rebase` merging strategy is desired.
+Merge commits in the code repositories are not allowed. The `rebase` strategy
+should be the only one available in the GitHub web UI.
+
+Is is, however, **strongly advised not to use GitHub web UI** to perform code
```
Comment (pietrushnic) [inline docs/dev-proc/source-code-structure.md:49]:
```suggestion
It is, however, **strongly advised not to use GitHub web UI** to perform code
```

**Review by pietrushnic — CHANGES_REQUESTED** [review]

File: docs/dev-proc/source-code-structure.md (hunk 1 above)
Comment (macpijan) [reply docs/dev-proc/source-code-structure.md:64]: @krystian-hebel Please push improvement to this PR if you beleive it makes sense

File: docs/dev-proc/source-code-structure.md (hunk 2 above)
Comment (macpijan) [reply docs/dev-proc/source-code-structure.md:49]: @krystian-hebel Can you fix it as well while changing git commands? Thanks!

File: docs/dev-proc/source-code-structure.md (hunk 1 above)
Comment (krystian-hebel) [reply docs/dev-proc/source-code-structure.md]: Done

File: docs/dev-proc/source-code-structure.md (hunk 2 above)
Comment (krystian-hebel) [reply docs/dev-proc/source-code-structure.md]: Done

**Pull request merged**